src/plugins/data/common/es_query/es_query/types.ts

```diff
@@ -6,9 +6,29 @@
  * Side Public License, v 1.
  */
 
-import { IFieldType } from '../../index_patterns';
+import type { estypes } from '@elastic/elasticsearch';
+
+export interface IFieldSubType {
+  multi?: { parent: string };
+  nested?: { path: string };
+}
+export interface FieldBase {
+  name: string;
+  type: string;
```

Comment:
Should add a comment that this is the kbnType, not the ES type

Reply:
Should this be typed differently?

Reply:
The name and type are accurate, just ambiguous to humans.

Reply:
hm, now that you mention it, it could be an enum but I suspect that would be a fair amount of work.